external-toolchain/recipes-sdk/sdk-env-external-toolchain/sdk-env-external-toolchain/external-codebench-toolchains.sh

```diff
@@ -1,7 +1,5 @@
 if [ -d "$scriptdir/../../../toolchains" ]; then
-    for d in "$scriptdir/../../../toolchains/"*/bin; do
-        if [ -d "$d" ]; then
-            PATH="$PATH:$d"
-        fi
+    for f in "$scriptdir/../../../toolchains/"*/environment-setup*; do
```

Comment:
Hmmmm. I believe we need to be a bit more cautious here. There are three things to note, at least this is what I'm seeing right now.

1. The ADE was built with a particular version of the toolchain. Currently we have 2020.07. If and when we install a newer toolchain for some reason, e.g. 2020.10, the script might pick that up in a way that it comes first in the PATH and when the binary name (e.g. aarch64-oe-linux-*) is same we go out of sync because our search is currently blind. We probably need to put the toolchain version in the env-setup and use that to at least identify the version and pick only that.
2. We should also consider using the CROSS_COMPILE or TARGET_PREFIX, whichever is more relevant, to limit the search further, less bloated PATH and 3.
3. The env-setup script from the toolchain adds the toolchain's sysroot/x86_64-oesdk-linux/usr/bin dir to the PATH which supersedes the ADE one the way we're doing it currently (this PR for now). This can cause issues as the mentioned directory contains some binaries which should ideally be used from the ADE rather than the toolchain
```
$ ls /data/Toolchains/codebench-linux-x64-12.1.0-arm/toolchains/aarch64-oe-linux.2020.07/sysroots/x86_64-oesdk-linux/usr/bin
2to3              eu-elflint             fdtdump         openssl         qemu-edid
2to3-3.8          eu-findtextrel         fdtget          prelink         qemu-ga
aarch64-oe-linux  eu-make-debug-archive  fdtoverlay      prelink-rtld    qemu-img
c_rehash          eu-objdump             fdtput          pydoc3          qemu-io
dtc               eu-ranlib              idle3           pydoc3.8        qemu-nbd
eu-ar             eu-stack               idle3.8         python3         qemu-pr-helper
eu-elfclassify    eu-strings             ivshmem-client  python3.8       qemu-system-aarch64
eu-elfcmp         eu-unstrip             ivshmem-server  python3.8.real  update-ca-certificates
eu-elfcompress    execstack              localedef       qemu-aarch64    update-mime-database
```
again we need to somehow only pick the PATH entry for the actual toolchain (from the above example: toolchains/aarch64-oe-linux.2020.07/sysroots/x86_64-oesdk-linux/usr/bin/aarch64-oe-linux - is that again TARGET_PREFIX/CROSS_COMPILE) and that should be it, no?

Reply:
If we assume we only ever build using the same toolchain that will be available in codebench, and that we'll never build an external ade from a non-external build, which is probably true, and we hardcode the particulars of the toolchain install path (it's defined in a variable in meta-codebench-toolchain that we don't have access to), we should be able to manage. i.e. `toolchains/${REAL_MULTIMACH_TARGET_SYS}.${SOURCERY_VERSION}/sysroots/${SDK_SYS}/usr/bin/${REAL_MULTIMACH_TARGET_SYS}`, and *append* that to PATH rather than prepending, but I could see that causing potential issues if there's ever a case where the ade lacks a particular nativesdk tool which is also available in the toolchain *and* on the host, it'd prefer the host to the toolchain binary in such a case. The only way to arrange for the ade bindir to be before the toolchain bindir and the latter before the host paths would be to alter how the env setup is generated rather than adding a environment-setup.d script.